plugins/storage/volume/default/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/CloudStackPrimaryDataStoreLifeCycleImpl.java

```diff
@@ -139,7 +139,13 @@ public DataStore initialize(Map<String, Object> dsInfos) {
         PrimaryDataStoreParameters parameters = new PrimaryDataStoreParameters();
 
         URI uri = null;
+        boolean multi = false;
         try {
+            String urlType = url.substring(0, 3);
```

Comment:
Should we not do this split after we have used the URI class and then just split the 'host' part?

Now you are doing manual string parsing on a string which we will then run through a library.